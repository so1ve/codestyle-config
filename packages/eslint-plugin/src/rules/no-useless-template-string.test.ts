import { run } from "./_test";
import rule, { RULE_NAME } from "./no-useless-template-string";

const valid = [
	"const a = '1';",
	'const a = "1"',
	// eslint-disable-next-line no-template-curly-in-string
	"const a = `1${b}`",
	"String.raw`str`",
	'`""`',
	"`''`",
];
const invalid = ["const a = `1`"];

run({
	name: RULE_NAME,
	rule,
	valid,
	invalid,
});
