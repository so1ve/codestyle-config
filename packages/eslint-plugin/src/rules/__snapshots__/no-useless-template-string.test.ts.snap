// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`no-useless-template-string > invalid > Invalid #0: const a = \`1\` 1`] = `"const a = "1""`;
